### PR TITLE
Sanitize logs as appropriate

### LINKED ISSUE #2644: master: Error: waitpid() returned unknown PID
### Environment

- [x] docker-compose
- [ ] kubernetes
- [ ] docker swarm

### Version

- Version: `master`

## Description

When deploying a standard mailu deployment, the imap container will log unknown pid error messages such as:
`mailu-imap-1  | Feb 02 08:49:07 master: Error: waitpid() returned unknown PID 533`

I still need to look into what actually logs this error.

## Replication Steps

1. Deploy Mailu.
2. Check the imap service logs (docker-compose logs imap )

## Observed behaviour

`master: Error: waitpid() returned unknown PID` errors are logged.

## Expected behaviour

No unknown pid errors are logs.

## Logs
This is only a ~20 minute period

```
imitri@mailutest:/mailu$ docker-compose logs imap | grep -ie "unknown pid"
mailu-imap-1  | Feb 02 08:34:20 master: Error: waitpid() returned unknown PID 25
mailu-imap-1  | Feb 02 08:34:54 master: Error: waitpid() returned unknown PID 47
mailu-imap-1  | Feb 02 08:35:28 master: Error: waitpid() returned unknown PID 65
mailu-imap-1  | Feb 02 08:36:03 master: Error: waitpid() returned unknown PID 85
mailu-imap-1  | Feb 02 08:36:37 master: Error: waitpid() returned unknown PID 102
mailu-imap-1  | Feb 02 08:37:11 master: Error: waitpid() returned unknown PID 122
mailu-imap-1  | Feb 02 08:37:45 master: Error: waitpid() returned unknown PID 141
mailu-imap-1  | Feb 02 08:38:19 master: Error: waitpid() returned unknown PID 159
mailu-imap-1  | Feb 02 08:38:53 master: Error: waitpid() returned unknown PID 177
mailu-imap-1  | Feb 02 08:39:27 master: Error: waitpid() returned unknown PID 194
mailu-imap-1  | Feb 02 08:40:01 master: Error: waitpid() returned unknown PID 214
mailu-imap-1  | Feb 02 08:40:35 master: Error: waitpid() returned unknown PID 230
mailu-imap-1  | Feb 02 08:41:09 master: Error: waitpid() returned unknown PID 250
mailu-imap-1  | Feb 02 08:41:44 master: Error: waitpid() returned unknown PID 268
mailu-imap-1  | Feb 02 08:42:18 master: Error: waitpid() returned unknown PID 286
mailu-imap-1  | Feb 02 08:42:52 master: Error: waitpid() returned unknown PID 304
mailu-imap-1  | Feb 02 08:43:26 master: Error: waitpid() returned unknown PID 323
mailu-imap-1  | Feb 02 08:44:00 master: Error: waitpid() returned unknown PID 343
mailu-imap-1  | Feb 02 08:44:34 master: Error: waitpid() returned unknown PID 359
mailu-imap-1  | Feb 02 08:45:08 master: Error: waitpid() returned unknown PID 379
mailu-imap-1  | Feb 02 08:45:42 master: Error: waitpid() returned unknown PID 398
mailu-imap-1  | Feb 02 08:46:16 master: Error: waitpid() returned unknown PID 415
mailu-imap-1  | Feb 02 08:46:50 master: Error: waitpid() returned unknown PID 433
mailu-imap-1  | Feb 02 08:47:25 master: Error: waitpid() returned unknown PID 452
mailu-imap-1  | Feb 02 08:47:59 master: Error: waitpid() returned unknown PID 472
mailu-imap-1  | Feb 02 08:48:33 master: Error: waitpid() returned unknown PID 499
mailu-imap-1  | Feb 02 08:49:07 master: Error: waitpid() returned unknown PID 533
mailu-imap-1  | Feb 02 08:49:41 master: Error: waitpid() returned unknown PID 556
mailu-imap-1  | Feb 02 08:50:15 master: Error: waitpid() returned unknown PID 572
mailu-imap-1  | Feb 02 08:50:49 master: Error: waitpid() returned unknown PID 587
mailu-imap-1  | Feb 02 08:51:23 master: Error: waitpid() returned unknown PID 604

```

### LINKED ISSUE #2764: rspamd crashes when hardened_malloc is enabled on some CPUs lacking AVX extensions
### Environment

- docker compose

### Version

- Version: `2.0.4`

## Description
While sending an email from an authenticated sender, the `rspamd_proxy` process crashes. This causes Postfix to reject the email. However, this doesn't always happen, but it happens frequently.

## Replication Steps
Sending email from an authenticated client.

## Observed behaviour
The active `rspamd_proxy` process crashes and a new one is spawned. As a result Postfix rejects the message with `4.7.1 Service unavailable - try again later`. The problem seems to have started after upgrading to Mailu 2.0.

## Expected behaviour
`rspamd_proxy` shouldn't crash.

## Logs
IPs and hostnames have been redacted / exchanged for documentation examples.
```
front      | 2023/04/11 17:10:47 [info] 11#11: *218 client [2001:0db8:85a3:0000:0000:8a2e:0370:7334]:56299 connected to [::]:587
postfix    | 2023-04-11T17:10:50.931564+00:00 5804f537147e postfix/smtpd[687]: connect from front.docker-mailu_default[10.0.0.9]
rspamd     | 2023-04-11 17:10:50 #166(rspamd_proxy) <10b8bc>; proxy; proxy_accept_socket: accepted milter connection from 10.0.0.3 port 42400
rspamd     | 2023-04-11 17:10:50 #166(rspamd_proxy) <10b8bc>; milter; rspamd_milter_process_command: got connection from 10.0.0.9:55478
rspamd     | 2023-04-11 17:10:50 #166(rspamd_proxy) <10b8bc>; proxy; proxy_milter_finish_handler: finished milter connection
rspamd     | 2023-04-11 17:10:50 #166(rspamd_proxy) <571832>; proxy; proxy_accept_socket: accepted milter connection from 10.0.0.3 port 42416
front      | 2023/04/11 17:10:50 [info] 11#11: *218 client logged in, client: 2001:0db8:85a3:0000:0000:8a2e:0370:7334 using starttls, server: [::]:587, login: "me@mydomain.com", upstream: 10.0.0.3:10025
postfix    | 2023-04-11T17:10:51.113207+00:00 5804f537147e postfix/smtpd[687]: 1B984E0145: client=rdns-name-from-isp.example.com[2001:0db8:85a3:0000:0000:8a2e:0370:7334], sasl_method=XCLIENT, sasl_username=me@mydomain.com
rspamd     | 2023-04-11 17:10:51 #166(rspamd_proxy) <571832>; milter; rspamd_milter_process_command: got connection from [2001:0db8:85a3:0000:0000:8a2e:0370:7334]:55478
postfix    | 2023-04-11T17:10:51.154829+00:00 5804f537147e postfix/cleanup[690]: 1B984E0145: message-id=<em36ea5242-dee7-4152-9bc4-b91f1541e85f@feda513d.com>
rspamd     | 2023-04-11 17:10:52 #1(main) rspamd_srv_handler: cannot read from worker's srv pipe connection closed; command = heartbeat
rspamd     | 2023-04-11 17:10:52 #1(main) <21dcb9>; main; rspamd_check_termination_clause: rspamd_proxy process 166 terminated abnormally with exit code 0 by signal: Illegal instruction but NOT created core file (throttled=no); core file limits: -1 current, -1 max
rspamd     | 2023-04-11 17:10:52 #1(main) <21dcb9>; main; rspamd_cld_handler: respawn process rspamd_proxy in lieu of terminated process with pid 166
postfix    | 2023-04-11T17:10:52.831005+00:00 5804f537147e postfix/cleanup[690]: warning: milter inet:antispam:11332: error writing command: Connection reset by peer
postfix    | 2023-04-11T17:10:52.832940+00:00 5804f537147e postfix/cleanup[690]: 1B984E0145: milter-reject: END-OF-MESSAGE from rdns-name-from-isp.example.com[2001:0db8:85a3:0000:0000:8a2e:0370:7334]: 4.7.1 Service unavailable - try again later; from=<me@mydomain.com> to=<destination@mydomain.com> proto=ESMTP helo=<[IPv6:2001:0db8:85a3:0000:0000:8a2e:0370:7334]>
postfix    | 2023-04-11T17:10:52.854248+00:00 5804f537147e postfix/smtpd[687]: disconnect from rdns-name-from-isp.example.com[2001:0db8:85a3:0000:0000:8a2e:0370:7334] ehlo=2 xclient=0/1 mail=1 rcpt=1 data=0/1 rset=1 quit=1 commands=6/8
front      | 2023/04/11 17:10:52 [info] 11#11: *218 proxied session done, client: 2001:0db8:85a3:0000:0000:8a2e:0370:7334 using starttls, server: [::]:587, login: "me@mydomain.com", upstream: 10.0.0.3:10025
```

The affected instance of Mailu is running on a netcup KVM server, that uses the following CPU profile:
```
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   40 bits physical, 48 bits virtual
CPU(s):                          2
On-line CPU(s) list:             0,1
Thread(s) per core:              1
Core(s) per socket:              2
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           13
Model name:                      QEMU Virtual CPU version 2.5+
Stepping:                        3
CPU MHz:                         2399.980
BogoMIPS:                        4799.96
Hypervisor vendor:               KVM
Virtualization type:             full
L1d cache:                       64 KiB
L1i cache:                       64 KiB
L2 cache:                        8 MiB
L3 cache:                        16 MiB
NUMA node0 CPU(s):               0,1
Vulnerability Itlb multihit:     KVM: Mitigation: VMX unsupported
Vulnerability L1tf:              Mitigation; PTE Inversion
Vulnerability Mds:               Mitigation; Clear CPU buffers; SMT Host state unknown
Vulnerability Meltdown:          Mitigation; PTI
Vulnerability Mmio stale data:   Unknown: No mitigations
Vulnerability Retbleed:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pse36 clflush mmx fxsr sse sse2 ht syscall nx lm rep_good nopl xtopology cpuid tsc_known_freq pni pclmulqdq ssse3 cx16 pcid sse4_1 sse4_2 x2apic popcnt aes xsave rdrand hypervisor lahf_lm abm c
                                 puid_fault invpcid_single pti ssbd ibrs ibpb invpcid md_clear arch_capabilities
```

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

enhancement

## What does this PR do?

- Sanitize logs as appropriate. 
- change the healthcheck of radicale to something less verbose
- disable hardened-malloc if we detect a processor not supporting the AVX extension set

Should we backport something like that? It could be argued it's a bugfix.

### Related issue(s)
- close #2644 
- close #2764
- #2541

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [ ] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by nextgens** [comment]
It probably needs to be tested better and tweaked before it's merged.

**Comment by nextgens** [comment]
And if we do merge it, should we backport it?

**Comment by Diman0** [comment]
We should not backport is since it breaks existing functionality (different knobs)

I don't see any changes to offer similar functionality to `POSTFIX_LOG_FILE`. 
As discussed in [PR1441](https://github.com/Mailu/Mailu/pull/1441), this functionality was added to provide an extra logging to file system option for postfix for kubernetes users. Apparently this is not possible via kubernetes configuration. Is this still possible?

**Comment by nextgens** [comment]
Yes we could probably keep rsyslog if we wanted to... I didn't know about why we had ``POSTFIX_LOG_FILE``.

If I kept the functionality the same, would we backport it?

**Comment by Diman0** [comment]
We discussed in the Mailu-Dev chat that `POSTFIX_LOG_FILE` is not required anymore. For the purpose to log all received/send messages to a log file (for legal reasons for example), you can
- Log postfix container output to systemd journald. This already happens for docker-compose and can also be configured for k8s.
- Connect rsyslogd to journald to import the unit log file of the postfix container.
- Use rsyslogd to log to file system.

Perhaps this is also possible by configuring journald? I'm not very familiar with custom journald configurations.

A faq entry will be created to inform users how to log to file system the postfix output.

**Comment by nextgens** [comment]
I've found a way to keep ``POSTFIX_LOG_FILE`` as is... so that we can backport the PR.

We can work on removing it in master afterwards.

**Comment by nextgens** [comment]
bors try

**Comment by Diman0** [comment]
bors try

File: core/nginx/start.py [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 import os
 import subprocess
+from socrate import system
+
+system.set_env(log_filters=r'could not be resolved \(\d\: [^\)]+\) while in resolving client address, client\: [^,]+, server: [^\:]+\:(25,110,143,587,465,993,995)$')
```
Comment (Diman0) [inline core/nginx/start.py:7]: This filter must be removed. It is used in our fail2ban config for banning bots. 
If we still choose to remove this from the logs, then the the fail2ban rule in faq.rst that uses this log message, must be omitted.

Personally I prefer still having this filter. It is an easy method to ban bad behaving bots.

**Review by Diman0 — CHANGES_REQUESTED** [review]
It filters out a log line that is used for banning bots via fail2ban. This breaks our fail2ban config in faq.rst.

File: core/nginx/start.py (hunk 1 above)
Comment (nextgens) [reply core/nginx/start.py:7]: Are you sure? This is the reverse DNS resolution failure line, not:

```
^\s?\S+ mailu\-front\[\d+\]: \S+ \S+ \[info\] \d+#\d+: \*\d+ client login failed: \"AUTH not supported\" while in http auth state, client: <HOST>, server:
```

Or are you saying that both get written in the same call and we should split/apply the regexp on each parts?

Comment (Diman0) [reply core/nginx/start.py:7]: I was wrong. It does work after all.

**Review by Diman0 — APPROVED** [review]
LGTM

**Pull request merged**